src/irc_bridge.py

```diff
@@ -51,7 +52,7 @@ def __init__(self, charm: ops.CharmBase, charm_state: CharmState):
         self.framework.observe(charm.on.collect_unit_status, self._on_collect_status)
 
     @property
-    def _pebble_service(self) -> typing.Any:
+    def _pebble_service(self) -> typing.Optional[PebbleService]:
```

Comment:
What would be the conditions where pebble service is `None` please? Making anything possibly `None` makes it's use more complicated because the user needs to check for whether it is `None` so if it can be avoided that it could be `None` that simplifies code a lot

Reply:
Has been refactored away.